transform/minifyhtml.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2016 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform
+
+import (
+	"bytes"
+	"regexp"
+
+	"github.com/tdewolff/minify"
+	"github.com/tdewolff/minify/css"
+	"github.com/tdewolff/minify/html"
+	"github.com/tdewolff/minify/js"
+	"github.com/tdewolff/minify/json"
+)
+
+var minifier *minify.M
+
+func init() {
+	m := minify.New()
+	m.AddFunc("text/css", css.Minify)
+	m.Add("text/html", &html.Minifier{
+		KeepDefaultAttrVals: true,
+		KeepWhitespace:      true,
+		KeepDocumentTags:    true,
+	})
+	m.AddFunc("text/javascript", js.Minify)
+	m.AddFuncRegexp(regexp.MustCompile("[/+]json$"), json.Minify)
+	minifier = m
+}
+
+// MinifyHTML minifies the HTML and the embedded CSS, JS and JSON.
+func MinifyHTML(ct contentTransformer) {
+	buf := bytes.NewReader(ct.Content())
+	if err := minifier.Minify("text/html", ct, buf); err != nil {
+		panic(err)
```

Comment:
Added a bit of clarification text.

Ideally we would modify `transform.trans` type definition to have an error returned from it. Would make error handling a lot cleaner. Thoughts, @bep?